BUILD.bazel

```diff
@@ -2394,37 +2394,148 @@ ray_cc_test(
 ray_cc_library(
     name = "object_manager",
     srcs = [
-        "src/ray/object_manager/chunk_object_reader.cc",
-        "src/ray/object_manager/common.cc",
-        "src/ray/object_manager/memory_object_reader.cc",
-        "src/ray/object_manager/object_buffer_pool.cc",
         "src/ray/object_manager/object_manager.cc",
-        "src/ray/object_manager/ownership_based_object_directory.cc",
-        "src/ray/object_manager/pull_manager.cc",
-        "src/ray/object_manager/push_manager.cc",
-        "src/ray/object_manager/spilled_object_reader.cc",
     ],
     hdrs = [
-        "src/ray/object_manager/chunk_object_reader.h",
-        "src/ray/object_manager/common.h",
-        "src/ray/object_manager/memory_object_reader.h",
-        "src/ray/object_manager/object_buffer_pool.h",
-        "src/ray/object_manager/object_directory.h",
         "src/ray/object_manager/object_manager.h",
-        "src/ray/object_manager/object_reader.h",
-        "src/ray/object_manager/ownership_based_object_directory.h",
-        "src/ray/object_manager/pull_manager.h",
+    ],
+    deps = [
+        ":chunk_object_reader",
+        ":object_buffer_pool",
+        ":plasma_store_server_lib",
+        ":pull_manager",
+        ":push_manager",
+    ],
+)
+
+ray_cc_library(
+    name = "push_manager",
+    srcs = [
+        "src/ray/object_manager/push_manager.cc",
+    ],
+    hdrs = [
         "src/ray/object_manager/push_manager.h",
-        "src/ray/object_manager/spilled_object_reader.h",
     ],
     deps = [
-        ":core_worker_lib",
-        ":gcs",
+        "//src/ray/common:id",
+        "//src/ray/common:ray_config",
+        ":stats_metric",
+    ],
+)
+
+ray_cc_library(
+    name = "pull_manager",
+    srcs = [
+        "src/ray/object_manager/pull_manager.cc",
+    ],
+    hdrs = [
+        "src/ray/object_manager/pull_manager.h",
+    ],
+    deps = [
+        ":ownership_based_object_directory",
         ":object_manager_rpc",
-        ":plasma_store_server_lib",
+        "//src/ray/util:counter_map",
+    ],
+)
+
+ray_cc_library(
+    name = "ownership_based_object_directory",
+    srcs = [
+        "src/ray/object_manager/ownership_based_object_directory.cc",
+    ],
+    hdrs = [
+        "src/ray/object_manager/ownership_based_object_directory.h",
+    ],
+    deps = [
+        ":gcs_client_lib",
```

Comment:
A few more deps should be included (include what you use).

Reply:
I also see opportunities to forward declare.
- instrumented_io_context
- GcsClient
- SubscriberInterface

Could be forward declared, in terms of ROI, I would suggest try `GcsClient`